src/delta/delta.c

```diff
@@ -392,19 +392,28 @@ static int enumerate_dir(
         return 0;
 }
 
-static int should_skip_prefix(const char* p) {
+static bool should_skip_path(const char *prefix, const char *suffix) {
 #if HAVE_SPLIT_USR
-        int r;
         _cleanup_free_ char *target = NULL;
+        const char *p;
+        char *dirname;
 
-        r = chase_symlinks(p, NULL, 0, &target);
-        if (r < 0)
-                return r;
+        dirname = strjoina(prefix, "/", suffix);
 
-        return !streq(p, target) && nulstr_contains(prefixes, target);
-#else
-        return 0;
+        if (chase_symlinks(dirname, NULL, 0, &target) < 0)
+                return false;
```

Comment:
Before, we could fail e.g. with EPERM. Now, such errors will be silently skipped. I don't think this change is warranted.

Reply:
We defer the error handling to the next caller which is going to fail anyways, Here we're only interested to see what is the target of the symlink but not in following it.